### PR TITLE
[SPARK-46035][BUILD] Upgrade zstd-jni to 1.5.5-10

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
The pr aims to upgrade zstd-jni from 1.5.5-7 to 1.5.5-10.

### Why are the changes needed?
This version solved some issues related to OSGi, such as https://github.com/luben/zstd-jni/commit/d5bc6a093ca909dd978fac47d35412d70b870da2,  https://github.com/luben/zstd-jni/commit/39c069776992812751402c7cf87cb3e89545b12d and https://github.com/luben/zstd-jni/commit/89c7dcb6b2c21c288e76b31dd567f0c71296facf.

The version compare as follows:
- https://github.com/luben/zstd-jni/compare/v1.5.5-7...v1.5.5-9

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Pass GitHub Actions

### Was this patch authored or co-authored using generative AI tooling?
No

### CODE REVIEW

**Comment by LuciferYang** [comment]
test 10

**Comment by dongjoon-hyun** [comment]
Thank you.

File: core/benchmarks/ZStandardBenchmark-results.txt [hunk 1]
```diff
@@ -2,26 +2,26 @@
...
 --------------------------------------------------------------------------------------------------------------------------------------
-Compression 10000 times at level 1 without buffer pool           2786           2787           2          0.0      278560.1       1.0X
-Compression 10000 times at level 2 without buffer pool           2831           2833           3          0.0      283091.0       1.0X
-Compression 10000 times at level 3 without buffer pool           2958           2959           2          0.0      295806.3       0.9X
-Compression 10000 times at level 1 with buffer pool               211            214           4          0.0       21145.3      13.2X
-Compression 10000 times at level 2 with buffer pool               253            255           1          0.0       25328.1      11.0X
-Compression 10000 times at level 3 with buffer pool               370            371           1          0.0       37046.1       7.5X
+Compression 10000 times at level 1 without buffer pool            657            659           2          0.0       65716.9       1.0X
```
Comment (LuciferYang) [inline core/benchmarks/ZStandardBenchmark-results.txt:9]: A new CPU model (AMD EPYC 7763) has been added to GA, but the benchmark results look significantly different from before. Let me see if we can retry the test on an Intel CPU.

Comment (dongjoon-hyun) [reply core/benchmarks/ZStandardBenchmark-results.txt:9]: Got it. For now, let's ignore this.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

File: core/benchmarks/ZStandardBenchmark-results.txt (hunk 1 above)
Comment (dongjoon-hyun) [reply core/benchmarks/ZStandardBenchmark-results.txt:9]: We have enough time for Apache Spark 4.0.0. So, we can find a way later. In addition, we will have more zstd-jni update.

**Comment by dongjoon-hyun** [comment]
Merged to master. Thank you, @LuciferYang .

**Comment by LuciferYang** [comment]
Thanks @dongjoon-hyun

**Pull request closed without merging**